pallets/pool-system/src/tranches.rs

```diff
@@ -1360,12 +1477,1610 @@ where
+			e_e_tranches
+				.combine_with_mut_residual_top([(200, 100), (200, 200), (300, 300)], |e, (i, r)| {
+					e.invest = i;
+					e.redeem = r;
+					Ok(())
+				})
+				.unwrap();
+
+			assert_eq!(
+				Ok(vec![100, 0, 0]),
+				e_e_tranches.supplies_with_fulfillment(&[
+					default_tranche_solution(),
+					default_tranche_solution(),
+					default_tranche_solution()
+				])
+			);
+
+			let mut e_e_tranches = default_epoch_tranches();
+
+			e_e_tranches
+				.combine_with_mut_residual_top([(200, 100), (200, 200), (300, 300)], |e, (i, r)| {
+					e.invest = i;
+					e.redeem = r;
+					Ok(())
+				})
+				.unwrap();
+
+			// Percentages for tranche solution taken into account for supplies
+			assert_eq!(
+				Ok(vec![50, 0, 0]),
+				e_e_tranches.supplies_with_fulfillment(&[
+					tranche_solution(Perquintill::from_percent(50), Perquintill::from_percent(50)),
+					default_tranche_solution(),
+					default_tranche_solution()
+				])
+			);
+
+			let mut e_e_tranches = default_epoch_tranches();
+
+			e_e_tranches
+				.combine_with_mut_residual_top(
+					[(200, 100, 100), (200, 100, 200), (300, 200, 300)],
+					|e, (i, s, r)| {
+						e.invest = i;
+						e.supply = s;
+						e.redeem = r;
+						Ok(())
+					},
+				)
+				.unwrap();
+
+			// Supplies updated correctly with pre-existing supply ammounts
+			// with invest amount doubling redeem for first tranch
+			assert_eq!(
+				Ok(vec![200, 100, 200]),
+				e_e_tranches.supplies_with_fulfillment(&[
+					default_tranche_solution(),
+					default_tranche_solution(),
+					default_tranche_solution()
+				])
+			);
+		}
+
+		#[test]
+		fn epoch_execution_calculate_weights_works() {
+			// Note: weight tuple containss: (investment_weight, redemption_weight)
+			// seniority of default tranches: 1, 2, 3
+			// expected weights: [
+			//  (highest_invest_weight, lowest_redemption_weight),
+			//  (mid_invest_weight, mid_redemption_weight),
+			//  (lowest_invest_weight, highest_redemption_Weight)
+			// ]
+			assert_eq!(
+				default_epoch_tranches().calculate_weights(),
+				vec![
+					(TrancheWeight::from(1000), TrancheWeight::from(10000)),
+					(TrancheWeight::from(100), TrancheWeight::from(100000)),
+					(TrancheWeight::from(10), TrancheWeight::from(1000000))
+				]
+			);
+
+			let mut e_e_tranches = default_epoch_tranches();
+
+			e_e_tranches
+				.combine_with_mut_residual_top([u32::MAX - 2, u32::MAX - 1, u32::MAX], |e, s| {
+					e.seniority = s;
+					Ok(())
+				})
+				.unwrap();
+			// Verify weights don't panic when calc would overflow
+			assert_eq!(
+				e_e_tranches.calculate_weights(),
+				vec![
+					(
+						TrancheWeight::from(u128::MAX),
+						TrancheWeight::from(u128::MAX)
+					),
+					(
+						TrancheWeight::from(u128::MAX),
+						TrancheWeight::from(u128::MAX)
+					),
+					(
+						TrancheWeight::from(u128::MAX),
+						TrancheWeight::from(u128::MAX)
+					)
+				]
+			)
+			// Verification of too many tranches unsurprisingly taking too long to run for tests
```

Comment:
Is there a benefit of commenting them out vs just removing them? Maybe opening an issue?

Reply:
Ah, yeah would be better to just remove, I'll remove it.  I'm not sure if it should have an issue, it's testing a pretty extreme edge case though.